FEATURE_MATRIX.md

```diff
@@ -1,61 +1,61 @@
 # Kernel Feature Matrix by Language
 
 ## AI Services
-| | C# | Python | Notes |
-|---|---|---|---|
-| TextGeneration                    | ✅ | ✅ | Example: Text-Davinci-003 |
-| TextEmbeddings                    | ✅ | ✅ | Example: Text-Embeddings-Ada-002 |
-| ChatCompletion                    | ✅ | ✅ | Example: GPT4, Chat-GPT |
-| Image Generation                  | ✅ | ❌ | Example: Dall-E |
+| | C# | Python | Java | Notes |
+|---|---|---|---|---|
+| TextGeneration                    | ✅ | ✅ | ✅ | Example: Text-Davinci-003 |
+| TextEmbeddings                    | ✅ | ✅ | ✅ | Example: Text-Embeddings-Ada-002 |
+| ChatCompletion                    | ✅ | ✅ | ❌ | Example: GPT4, Chat-GPT |
+| Image Generation                  | ✅ | ❌ | ❌ | Example: Dall-E |
 
 ## AI Service Endpoints
-| | C# | Python | Notes |
-|---|---|---|---|
-| OpenAI                            | ✅ | ✅ | |
-| AzureOpenAI                       | ✅ | ✅ | |
-| Hugging Face Inference API        | 🔄 | ❌ | Coming soon to Python, not all scenarios are covered for .NET |
-| Hugging Face Local                | ❌ | ✅ | |
-| Custom                            | ✅ | 🔄 | Requires the user to define the service schema in their application |
+| | C# | Python | Java| Notes |
+|---|---|---|---|---|
+| OpenAI                            | ✅ | ✅ | ✅ | |
+| AzureOpenAI                       | ✅ | ✅ | ✅ | |
+| Hugging Face Inference API        | 🔄 | ❌ | ❌ | Coming soon to Python, not all scenarios are covered for .NET |
+| Hugging Face Local                | ❌ | ✅ | ❌ | |
+| Custom                            | ✅ | 🔄 | ❌ | Requires the user to define the service schema in their application |
 
 ## Tokenizers
-| | C# | Python | Notes |
-|---|---|---|---|
-| GPT2                              | ✅ | ✅ | |
-| GPT3                              | ✅ | ❌ | |
+| | C# | Python | Java | Notes |
+|---|---|---|---|---|
+| GPT2                              | ✅ | ✅ | ❌ | |
+| GPT3                              | ✅ | ❌ | ❌ | |
 | tiktoken                          | 🔄 | ❌ | Coming soon to Python and C#. Can be manually added to Python via `pip install tiktoken` |
 
 ## Core Skills
-| | C# | Python | Notes |
-|---|---|---|---|
-| TextMemorySkill                   | ✅ | ✅ | |
-| ConversationSummarySkill          | ✅ | ❌ | |
-| FileIOSkill                       | ✅ | ✅ | |
-| HttpSkill                         | ✅ | ✅ | |
-| MathSkill                         | ✅ | ❌ | |
-| TextSkill                         | ✅ | ✅ | |
-| TimeSkill                         | ✅ | ✅ | |
+| | C# | Python | Java | Notes |
+|---|---|---|---|---|
+| TextMemorySkill                   | ✅ | ✅ | ✅ | |
```

Comment:
will need to ask David on this but I believe that may still be considered wip

Reply:
TextMemorySkill?